### PR TITLE
System_packages: speed up packages endpoint

### PR DESCRIPTION
**Pull request by semtexzv**
The indexes in packages table forced us to do 2 joins before aggregation. this adds name_id into the table, and allows us perform aggregation before joins, which should greatly speed up the packages endpoint.

### CODE REVIEW

File: database_admin/migrations/047_system_package_reverse_idx.down.sql [hunk 1]
```diff
@@ -0,0 +1 @@
+DROP INDEX IF EXISTS system_package_package_system_idx;
```
Comment (Josca) [inline database_admin/migrations/047_system_package_reverse_idx.down.sql:1]: This should contain also recreating of previous index and dropping of name_id column.

File: database_admin/schema/create_schema.sql [hunk 2]
```diff
@@ -883,15 +883,20 @@ CREATE UNIQUE INDEX IF NOT EXISTS package_latest_cache_pkey ON package_latest_ca
 
 CREATE TABLE IF NOT EXISTS system_package
 (
-    rh_account_id INT NOT NULL REFERENCES rh_account,
-    system_id     INT NOT NULL,
-    package_id    INT NOT NULL REFERENCES package,
+    rh_account_id INT                                  NOT NULL REFERENCES rh_account,
```
Comment (Josca) [inline database_admin/schema/create_schema.sql:886]: I think that cleaner would be to list columns in table definition in the same order as used in index, so: rh_account_id, name_id, package_id, system_id. The same for definition of gorm struct.

**Review by Josca — CHANGES_REQUESTED** [review]
Good, I hope it will help us. But now there are some conflicts to resolve.

File: database_admin/migrations/047_system_package_reverse_idx.down.sql (hunk 1 above)
Comment (semtexzv) [reply database_admin/migrations/047_system_package_reverse_idx.down.sql]: We're not dropping previous indexes

File: database_admin/schema/create_schema.sql (hunk 2 above)
Comment (semtexzv) [reply database_admin/schema/create_schema.sql:939]: Adding a column adds it at the end of the list, We can't just reorder table columns.

File: manager/controllers/packages.go [hunk 3]
```diff
@@ -46,7 +45,7 @@ func packagesQuery(c *gin.Context, acc int) (*gorm.DB, error) {
 	subQ := database.SystemPackages(database.Db, acc).
 		Select(queryItemSelect).
 		Where("sp.stale = false").
-		Group("p.name_id, pn.name")
+		Group("spkg.name_id")
```
Comment (Josca) [inline manager/controllers/packages.go:48]: we can improve it: https://github.com/semtexzv/patchman-engine/pull/2

**Review by Josca — CHANGES_REQUESTED** [review]

**Review by Josca — APPROVED** [review]
Great, but we should merge and deploy "culling fixes PR" before this one.

**Pull request merged**